### ISSUE TITLE
User Defined Procedure - Error while building

### ISSUE THREAD

**Issue by JakobWyatt-VGW** [tuddman/clj-neo4j-procedures#1]
Does this work for user defined procedures (not functions) in Clojure?
After removing the #_ (ignore next form) macros in procedures.clj, I get an error while building: `Unable to resolve classname: CljObjectResult`.
Did you ever manage to get this to work?
Thanks

**Comment by tuddman** [comment]
I don't believe so.  Or at least I don't remember procedures ever working.  Functions, yes.

It's been awhile since I've wanted/needed this library to be functional.   Sorry if that's not much help.

I seem to recall that my block was around learning the correct usage of annotations, which Neo4j relies(ed?) on, and clojure has some ability to use.   I put some random notes [here](https://github.com/tuddman/clj-neo4j-procedures/blob/master/doc/notes.md)

If you manage to get something going, please share it back and I'll make a point to update this library to a usable state.

**Comment by JakobWyatt-VGW** [comment]
Thanks for letting me know - I fiddled around for a day or so trying to get clojure to work, but didn't really end up getting anywhere. Annotations were what killed me too.
Ended up just using Scala instead - I'll probably have another look at this if I have any free time and nothing to do.

Will be sure to make a pull request if I get anything working.
Thanks

**Comment by tuddman** [comment]
I've always thought that, if you get procedures + functions to work properly from within clojure, then there's nothing really stopping you from building an entire application that could be 'exposed' as Neo4j plugin.

This would work especially well if the heart of the application was doing complex graph analysis/processing, etc.

So rather than have your application need a lot of boilerplate ORM-style calls between the app & DB, as has been the pattern since forever ... why not just put the entire application colocated to the DB and be done with it?          
Yes you are coupling the application to the DB, but again if the purpose for the app is to in some way(s) richly interact with a graph database and all of the data it contains, then why not? :)

**Comment by tuddman** [comment]
shot in the dark here...

reaching out to @jexp @mneedham @ducky427 @chrisbetz.    as I've come across your profiles in the (small) overlap of people familiar with both `clojure` +` neo4j`

my ask, plain and simple:      

Would any of you care to take a crack at helping 'solve' building Neo4j Functions and/or Procedures ... in Clojure?      I've been tripped up around Annotations and general Java Interop.

My original aim for this library was, in still is, to provide an easy way for the Clojure community to leverage, and build up, Neo4j  plugins, that can obviously be packaged as jar's and shared with the broader Neo4j community.  Would love to find out if it is indeed possible.

Any help or even comments are most appreciated!    Thanks, and hope everyone is well.

**Comment by chrisbetz** [comment]
Hi @tuddman,

currently I'm busy working on other stuff, but I like the idea in general. So, what I can offer to you: If you have a concrete test case not running (and isolated enough to understand it), I can try to have a look at it. But I cannot promise anything wrt working on another project in my free time.
